pandas/core/common.py

```diff
@@ -182,7 +182,8 @@ def _maybe_box_datetimelike(value):
 
 
 def is_bool_indexer(key):
-    if isinstance(key, (ABCSeries, np.ndarray)):
+    from pandas.core.index import Index
```

Comment:
you can use ``ABCIndexClass`` here instead (and import were ABCSeries is imported)